tests/test_dose.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""unittest cases for dose."""
+# test_dvhdose.py
+# Copyright (c) 2016-2020 Aditya Panchal
+# Copyright (c) 2020 Dan Cutright
+
+
+from __future__ import division
+import unittest
+import os
+from dicompylercore import dicomparser, dose
+
+try:
+    from pydicom.dataset import Dataset
+    from pydicom.sequence import Sequence
+    from pydicom import read_file as read_dicom
+except ImportError:
+    from dicom.dataset import Dataset
+    from dicom.sequence import Sequence
+    from dicom import read_file as read_dicom
+from numpy import arange, zeros
+from numpy.testing import (
+    assert_array_almost_equal,
+    assert_array_equal,
+    assert_raises,
+)
+import warnings
+
+basedata_dir = "tests/testdata"
+example_data = os.path.join(basedata_dir, "example_data")
+
+
+def assert_array_not_equal(arr_1, arr_2):
+    return assert_raises(AssertionError, assert_array_equal, arr_1, arr_2)
+
+
+def assert_array_not_almost_equal(arr_1, arr_2):
+    return assert_raises(
+        AssertionError, assert_array_almost_equal, arr_1, arr_2
+    )
+
+
+class TestDose(unittest.TestCase):
+    """Unit tests for Dose calculations."""
+
+    def setUp(self):
+        """Setup files for common case testing."""
+        self.rtdose_dcm = os.path.join(example_data, "rtdose.dcm")
+        self.rtdose = dicomparser.DicomParser(self.rtdose_dcm)
+
+        self.dvhs = self.rtdose.GetDVHs()
+
+        self.dosegrid = dose.DoseGrid(self.rtdose_dcm)
+
+    def test_modality_check(self):
+        """Test non-RTDOSE raises AttributeError"""
+        ds = dicomparser.DicomParser(self.rtdose_dcm).ds
+        ds.Modality = "RTPLAN"
+        with self.assertRaises(AttributeError):
+            dose.DoseGrid(ds)
+
+    def test_shape(self):
+        """Test dose grid shape extraction from the DICOM data."""
+        shape = self.dosegrid.shape
+
+        # x-dimension
+        self.assertEqual(shape[0], 194)
+        # y-dimension
+        self.assertEqual(shape[1], 129)
+        # z-dimension
+        self.assertEqual(shape[2], 98)
+
```

Comment:
A while since I used unittest, but can't the whole thing be compared together?

```python
self.assertEqual(shape, (194, 129, 98))
```

Reply:
Good thought.  I had to use assert_array_equal for a couple, but otherwise good to go now.